packages/graphql/lib/src/link/web_socket/link_web_socket.dart

```diff
@@ -23,15 +22,13 @@ class WebSocketLink extends Link {
   /// Creates a new [WebSocketLink] instance with the specified config.
   WebSocketLink(
       {@required this.url,
-      this.headers,
```

Comment:
How about, instead of removing it entirely, we:
* mark the header argument as as deprecated
* throw an error `if (dart.library.html)`
* warn when it's used `if (dart.library.io)` similar to the legacy payload

```dart
if (headers != null) {
  assert(
    !dart.library.html,
    'Cannot set websocket headers with dart:html websockets. '
    'If these are for authentication, another approach must be used, '
    'such as initPayload.',
  );
  print(
    'WARNING: Using direct websocket headers which will be removed soon, '
    'as it is incompatable with dart:html. '
    'If you need this direct header access, '
    'please comment on this PR with details on your usecase: '
    'https://github.com/zino-app/graphql-flutter/pull/323',
  );
  concatHeaders.addAll(headers);
}
```

Reply:
agree, will make appropriate change.

Reply:
please review the latest change, I have this assertion in constructor instead.